File: app/src/main/java/com/mahesh/mvpsample/main/MainPresenter.java
package com.mahesh.mvpsample.main;

public class MainPresenter implements MainContract.Presenter {
    private MainContract.View view;

    public MainPresenter(MainContract.View view) {
        this.view = view;
    }

    @Override
    public void create() {

    }

    @Override
    public Boolean isViewAvailable() {
        return view != null;
    }

    @Override
    public void resume(MainContract.View view) {
        this.view = view;
    }

    @Override
    public void pause() {
        view = null;
    }

    @Override
    public void clickMeButtonClicked() {
        if (isViewAvailable()) {
            view.openDetailScreen();
        }
    }
}

File: app/src/main/java/com/mahesh/mvpsample/main/MainPresenter.kt
package com.mahesh.mvpsample.main

class MainPresenter(private var view: MainContract.View?) : MainContract.Presenter {
    override fun create() {}
    override val isViewAvailable: Boolean
        get() = view != null

    override fun resume(view: MainContract.View?) {
        this.view = view
    }

    override fun pause() {
        view = null
    }

    override fun clickMeButtonClicked() {
        if (isViewAvailable) {
            view!!.openDetailScreen()
        }
    }
}
